omp/base/index_set_kernels.cpp

```diff
@@ -63,21 +65,18 @@ namespace index_set {
 
 template <typename IndexType>
 void to_global_indices(std::shared_ptr<const DefaultExecutor> exec,
-                       const IndexType index_space_size,
-                       const Array<IndexType>* subset_begin,
-                       const Array<IndexType>* subset_end,
-                       const Array<IndexType>* superset_indices,
-                       Array<IndexType>* decomp_indices)
+                       const IndexType num_subsets,
+                       const IndexType* subset_begin,
+                       const IndexType* subset_end,
+                       const IndexType* superset_indices,
+                       IndexType* decomp_indices)
 {
-    auto indices = decomp_indices->get_data();
-    auto num_subsets = superset_indices->get_num_elems() - 1;
-    auto ss_indices = superset_indices->get_const_data();
 #pragma omp parallel for
     for (size_type subset = 0; subset < num_subsets; ++subset) {
-        for (size_type i = 0; i < ss_indices[subset + 1] - ss_indices[subset];
-             ++i) {
-            indices[ss_indices[subset] + i] =
-                subset_begin->get_const_data()[subset] + i;
+        for (size_type i = 0;
+             i < superset_indices[subset + 1] - superset_indices[subset]; ++i) {
+            decomp_indices[superset_indices[subset] + i] =
+                subset_begin[subset] + i;
         }
```

Comment:
could be turned into the more common form
```suggestion
        IndexType local_i{};
        for (auto i = superset_indices[subset]; i < superset_indices[subset + 1]; ++i) {
            decomp_indices[i] = local_i + subset_begin[subset];
            local_i++;
        }
```